src/core/defs.jl

```diff
@@ -99,6 +99,7 @@ istype(T::DataType) = (pair -> pair.second isa T)
 
 # Namespace filter functions return dicts of values for the given type.
 # N.B. only composites hold other comps in the namespace.
+components(model::Model, comp_name::Symbol) = components(compdef(model, comp_name))
 components(obj::AbstractCompositeComponentDef) = filter(istype(AbstractComponentDef), obj.namespace)
```

Comment:
I'm thinking that for this it might be good to open it up so that it acts on `obj::AbstractComponentDef` and then if it's called on a leaf component, which is a `ComponentDef`, it returns an empty array. In the future this seems like a cleaner way to handle recursive type calls ... @ckingdon95 thoughts on that?